frontend/src/createStream/StreamingTable.js

```diff
@@ -0,0 +1,127 @@
+import React from 'react';
+import {
+    Grid
+} from "@material-ui/core"
+
+import axios from "axios";
+
+
+import SelectDevicesTable from "./SelectDevicesTable";
+import StreamButton from "../General/Buttons/StreamButton";
+
+export default class StreamingTable extends React.Component {
+
+    constructor(props) {
+        super(props)
+        this.state = {
+            senders: [],
+            receivers: [],
+            selectedSender: {},
+            selectedSenderID: "",
+            selectedReceiver: {},
+            selectedReceiverID: ""
+        }
+
+        this.dataSource = props.dataSource;
+        this.handleSendersChange = this.handleSendersChange.bind(this);
+        this.handleReceiversChange = this.handleReceiversChange.bind(this);
+
+        this.onSenderSelected = this.onSenderSelected.bind(this);
+        this.onReceiverSelected = this.onReceiverSelected.bind(this);
+
+        this.handleSubmit = this.handleSubmit.bind(this);
+    }
+
+    componentDidMount() {
+        this.dataSource.getSenders(this.handleSendersChange);
+        this.dataSource.getReceivers(this.handleReceiversChange);
+    }
+
+    handleSendersChange(senders) {
+        this.setState({
+            senders: senders
+        });
+    }
+
+    handleReceiversChange(receivers) {
+        this.setState({
+            receivers: receivers
+        });
+    }
+
+    onSenderSelected(selectedSender) {
+        var data = selectedSender.target.value.split("_");
+        var sender = this.state.senders[data[0]];
+
+        this.setState({
+            selectedSender: sender,
+            selectedSenderID: data[2]
+        })
+    }
+
+    onReceiverSelected(selectedReceiver) {
+        var data = selectedReceiver.target.value.split("_");
+        var receiver = this.state.receivers[data[0]];
+
+        this.setState({
+            selectedReceiver: receiver,
+            selectedReceiverID: data[2]
+        })
+
+    }
+
+    handleSubmit(event) {
+
+        if (this.state.selectedReceiverID === "" || this.state.selectedSenderID === "") {
+            console.log("Please select a sender and a receiver");
+        }
+        else {
+            axios.post("http://localhost:8080/stream", {
+                inputChannelId: this.state.selectedSenderID,
+                outputChannelId: this.state.selectedReceiverID
```

Comment:
can this be abstracted in a streamApi file?
Is this a new endpoint created by the new PR by francois?

Reply:
Yeah, It's in anticipation for francois's PR 
I don't know much abt streamAPI files but I'll look into it

Reply:
its just because i saw that you have the device requests in a file called DeviceApi, so maybe u can do something like that for stream endpoints